server/dataset/dataset.py

```diff
@@ -72,7 +72,7 @@ def get_embedding_array(self, ename, dims=2):
         pass
 
     @abstractmethod
-    def get_X_array(self, obs_mask=None, var_mask=None):
+    def get_X_array(self, obs_mask=None, var_mask=None, allow_sparse_return=False):
```

Comment:
There does not seem to be a test case for this new feature (allow_sparse_return). Please add (and check codecov to make sure you hit all the corner cases)